agent/agent.go

```diff
@@ -539,8 +598,13 @@ func (a *Agent) reloadWatches(cfg *Config) error {
 func (a *Agent) consulConfig() (*consul.Config, error) {
 	// Start with the provided config or default config
 	base := consul.DefaultConfig()
+
+	// a.config.ConsulConfig, if set, is a partial configuration for the
+	// consul server or client. Therefore, clone and augment it but
+	// don't use it as base directly.
 	if a.config.ConsulConfig != nil {
-		base = a.config.ConsulConfig
+		base = new(consul.Config)
```

Comment:
This is just style. Personally, I prefer the `&consul.Config{}` style as well but the rest of the code seems to go with `new(xxx)`. So I stuck with that.